src/skmatter/preprocessing/_data.py

```diff
@@ -524,7 +524,11 @@ def fit(self, Knm, Kmm, y=None, sample_weight=None):
         if self.with_trace:
             Knm_centered = Knm - self.K_fit_rows_
 
-            Khat = Knm_centered @ np.linalg.pinv(Kmm, self.rcond) @ Knm_centered.T
+            # The following is more correctly written as Knm @ Kmm^{-1} @ Knm.T
+            # but has been changed to Knm.T @ Knm @ Kmm^{-1} to avoid the memory
+            # overload often caused by storing n x n matrices. This is fine
+            # for the following trace, but should not be used for other operations.
+            Khat = Knm_centered.T @ Knm_centered @ np.linalg.pinv(Kmm, self.rcond)
```

Comment:
Maybe it is more failsafe to write

```suggestion
            Khat_trace = np.trace(Knm_centered.T @ Knm_centered @ np.linalg.pinv(Kmm, self.rcond))
```
and use this later on. Because technically this is not Khat anymore. But no strong opinion